pilot/pkg/model/validation.go

```diff
@@ -1222,22 +1222,21 @@ func ValidateDestinationPolicy(msg proto.Message) error {
 
 // ValidateProxyAddress checks that a network address is well-formed
 func ValidateProxyAddress(hostAddr string) error {
-	colon := strings.Index(hostAddr, ":")
-	if colon < 0 {
-		return fmt.Errorf("':' separator not found in %q, host address must be of the form <DNS name>:<port> or <IP>:<port>",
-			hostAddr)
+	host, p, err := net.SplitHostPort(hostAddr)
+	if err != nil {
+		return fmt.Errorf("unable to split %q: %s", hostAddr, err.Error())
 	}
-	port, err := strconv.Atoi(hostAddr[colon+1:])
+	port, err := strconv.Atoi(p)
```

Comment:
in theory it could be a service name (as in /etc/services not istio service) but that’s prexisting/not ipv6 issue

Reply:
Right, I'm trying to keep in line with the current design, expecting an address and port number.